### ISSUE TITLE
automount /jffs

### ISSUE THREAD

**Issue by sy-uname** [lukasz1992/asuswrt-xiaomi#121]
Hello.
I have Mi Router 4A giga with firmware AsusWRT	3.0.0.4.382_52302-geac6f28
The Administration-> System menu do not show the option "Enable JFFS custom scripts and configs"
After boot system 
`admin@RT-MIR4A:/tmp/home/root# mount`
`rootfs on / type rootfs (rw)`
`/dev/root on / type squashfs (ro,relatime)`
`proc on /proc type proc (rw,relatime)`
`tmpfs on /tmp type tmpfs (rw,relatime)`
`devfs on /dev type tmpfs (rw,noatime)`
`sysfs on /sys type sysfs (rw,relatime)`
`devpts on /dev/pts type devpts (rw,relatime,mode=600)`
After hand mount `mount -t jffs2 /dev/mtdblock5 /jffs` directory staying available for writing? but only until reboot.
How fix that?

**Comment by lukasz1992** [comment]
Is there anything in log after you mount manually?
Was it working before?

I no longer have the device to test :(

**Comment by lukasz1992** [comment]
Please show me output of `cat /proc/mtd` and check if command `mtd-erase -d jffs2` succeeds.
Please also share output of `nvram show | grep jffs | sort`

**Comment by sy-uname** [comment]
>Was it working before?
> 
No, it's fresh installation firmware, previous was original firmware Xiaomi.
> log after you mount manually
>
`grep -i jffs /tmp/syslog.log` 
`May  5 08:05:09 kernel: 0x000000df0000-0x000000ff0000 : "jffs2"`
`May  5 08:05:33 jffs2: valid logs(1)`
No error or other messages
> mtd-erase -d jffs2
>
`admin@RT-MIR4A:/tmp/home/root# mtd-erase -d jffs2`
`Erasing 0x0 - 0xffff`
...
`Erasing 0x1f0000 - 0x1fffff`
`"jffs2" successfully erased.`
>cat /proc/mtd 
>
`dev:    size   erasesize  name`
`mtd0: 00050000 00010000 "Bootloader"`
`mtd1: 00010000 00010000 "nvram"`
`mtd2: 00010000 00010000 "Factory"`
`mtd3: 00c70000 00010000 "linux"`
`mtd4: 00aea080 00010000 "rootfs"`
`mtd5: 00200000 00010000 "jffs2"`
`mtd6: 01000000 00010000 "ALL"`
>nvram show | grep jffs | sort
>
`size: 26385 bytes (35055 left)`
`jffs2_enable=1`
`jffs2_format=0`
`jffs2_on=1`
`jffs2_scripts=1`
`jffs2_size=2097152`

**Comment by lukasz1992** [comment]
Looks good. Let's try to enable debug, please:

```
nvram set debug_cprintf=1
nvram set debug_cprintf_file=1
nvram commit
reboot
```
After reboot please login, check jffs2 is not mounted, and please send me output of `/tmp/cprintf`

**Comment by sy-uname** [comment]
reboot, jffs2 is not mounted, file /tmp/cprintf was created:
[cprintf.zip](https://github.com/lukasz1992/asuswrt-xiaomi/files/8238111/cprintf.zip)

**Comment by lukasz1992** [comment]
I would try three things:
- Try to set `nvram set jffs2_format=1; nvram commit`
- Factory reset
- Modify source code - replace src/router/rc/jffs2.c, line 143:
```
return;
```
with
```
;
```
recompile source code and see if it helps

**Comment by sy-uname** [comment]
>- Try to set nvram set jffs2_format=1; nvram commit; reboot
>
I already did it, but it didn't help.
>- 
>
